### PR TITLE
[Integrations-Testing] Test creation docs

### PR DESCRIPTION
**Pull request by KSGulin**

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]

File: tests/integrations/base_tester.py [hunk 1]
```diff
@@ -220,6 +220,10 @@ class BaseIntegrationTester:
     """
     Class from which integration test-holding classes should inherit. Tests defined here
     need to be implemented for each integration on the integration level.
+
+    All tests are expected to follow the name convention test_{stage}_{name} where
+    stage is train, export, or deploy and name is a unique name to describe the test.
```
Comment (dbogunowicz) [inline tests/integrations/base_tester.py:225]:
```suggestion
    stage is `train`, `export`, or `deploy` and name is a unique name to describe the test.
```

File: tests/integrations/base_tester.py [hunk 2]
```diff
@@ -220,6 +220,10 @@ class BaseIntegrationTester:
...
+    This naming convention is used and enforced within the decorator skip_inactive_stage
```
Comment (dbogunowicz) [inline tests/integrations/base_tester.py:226]:
```suggestion
    This naming convention is used and enforced within the decorator `@skip_inactive_stage`
```

File: tests/integrations/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+**pre_args:** [WIP][Optional] Allows for the addition of CLI commands to pre-pend to the run command. Most common use case is for setting environment variables.
+
+**command_args:** [Optional/Required] The arguments to be passed to each stage of the run. These arguments are converted into CLI commands for runtime (example below). Supported args are defined by the `arg class` implementations for the integration. Thus if the corresponding class defines required arguments, then the `command_args` field is required. Field names provided in the config that don't correspond to supported arguments in the code are flagged with an error to prevent user error. Example of arg conversions from config to CLI command <br>
+`'epochs: 5' -> '--epochs 5'` <br>
+`'dynamic: True' -> '--dynamic'`
+
+**test_args:** These arguments provide information to the tests on what quantities and qualities of the run output to test. These can integration specific, but right now the following set of test args are supported across the integrations:
+- Train
+    - target_name: Name of the metric to test
```
Comment (dbogunowicz) [inline tests/integrations/README.md:70]:
```suggestion
    - `target_name`: Name of the metric to test
```

File: tests/integrations/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    - target_mean: Mean value of the named metric
```
Comment (dbogunowicz) [inline tests/integrations/README.md:71]:
```suggestion
    - `target_mean`: Mean value of the named metric
```

File: tests/integrations/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    - target_std: Acceptable range of deviation from the target mean for the named metric
```
Comment (dbogunowicz) [inline tests/integrations/README.md:72]:
```suggestion
    - `target_std`: Acceptable range of deviation from the target mean for the named metric
```

File: tests/integrations/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+- Export
+    - target_model: zoo stub for a model to test the onnx model produced by the run against
```
Comment (dbogunowicz) [inline tests/integrations/README.md:74]:
```suggestion
    - `target_model`: zoo stub for a model to test the onnx model produced by the run against
```

File: tests/integrations/README.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+**pre_args:** [WIP][Optional] Allows for the addition of CLI commands to pre-pend to the run command. Most common use case is for setting environment variables.
+
+**command_args:** [Optional/Required] The arguments to be passed to each stage of the run. These arguments are converted into CLI commands for runtime (example below). Supported args are defined by the `arg class` implementations for the integration. Thus if the corresponding class defines required arguments, then the `command_args` field is required. Field names provided in the config that don't correspond to supported arguments in the code are flagged with an error to prevent user error. Example of arg conversions from config to CLI command <br>
+`'epochs: 5' -> '--epochs 5'` <br>
+`'dynamic: True' -> '--dynamic'`
+
+**test_args:** These arguments provide information to the tests on what quantities and qualities of the run output to test. These can integration specific, but right now the following set of test args are supported across the integrations:
```
Comment (dbogunowicz) [inline tests/integrations/README.md:68]:
```suggestion
**test_args:** These arguments provide information to the tests on what quantities and qualities of the run output to test. These can integration specific, but right now the following set of `test args` are supported across the integrations:
```

File: tests/integrations/README.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+## Config Fields
+**cadence:** [Required] Defines how often the test should be run. Supported values are `commit`, `nightly`, and `weekly`. Upon testing, config files with the desired cadence will be be fetched, ran, and tested.
+
+**abridged_mode:** [Optional] When true, the length of the run will be automatically shortened to a standardized abridged version. e.g. for training, a default, low value of maximum training and eval steps will be set. This allows for testing of long runs on a higher cadence. Naturally, metrics can not be tested for abridged runs. Defaults to False when not set
+
+**Stage Name:** [>1 Required] Supported stage names are `train`, `export`, and `deploy` [WIP]. None are individually required, but at least one must be present in the config. 
+
+**pre_args:** [WIP][Optional] Allows for the addition of CLI commands to pre-pend to the run command. Most common use case is for setting environment variables.
```
Comment (dbogunowicz) [inline tests/integrations/README.md:62]:
```suggestion
**pre_args:** [WIP][Optional] Allows for the addition of CLI commands to prepend to the run command. The most common use case is for setting environment variables.
```

File: tests/integrations/README.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    pre_args: sample_pre_arg
+    command_args:
+        model_path: path/to/model
+        data: path/to/data
+    test_args:
+        test_arg: test_value
+```
+
+## Config Fields
+**cadence:** [Required] Defines how often the test should be run. Supported values are `commit`, `nightly`, and `weekly`. Upon testing, config files with the desired cadence will be be fetched, ran, and tested.
```
Comment (dbogunowicz) [inline tests/integrations/README.md:56]:
```suggestion
**cadence:** [Required] Defines how often the test should be run. Supported values are `commit`, `nightly`, and `weekly`. Upon testing, config files with the desired cadence will be fetched, ran, and tested.
```

**Review by dbogunowicz — COMMENTED** [review]
Couple of nits, mostly for better style of the doc.

**Comment by KSGulin** [comment]
@dbogunowicz thanks for upgrading the style. Ready for re-review

**Review by bfineran — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]

**Pull request merged**